scripts/bash_utils.sh

```diff
@@ -40,3 +23,16 @@ function setup_python {
   fi
   echo ${python_bin}
 }
+
+function install_py_packages {
+  local deps="black==$BLACK_VERSION flake8==$FLAKE8_VERSION clang-format==$CLANG_FORMAT_VERSION clang-tidy==$CLANG_TIDY_VERSION"
+
+  local py=$1
+  if [[ $(${py} -m pip list | grep -c "black\|flake8\|clang-format\|clang-tidy") -ne 4 ]]; then
+    if [[ $(${py} -c 'import sys; print(int(sys.base_prefix != sys.prefix or hasattr(sys, "real_prefix")))') -eq 1 ]]; then
+      ${py} -m pip install ${deps}
```

Comment:
nit: From one perspective, it is nice to have these versions pinned right where they are used, but at the same time it looks quite scary. What do you think about
- using `requirements.txt`
- using external package managers that are able to handle different python versions and handle local envs in a single place like https://github.com/astral-sh/uv.

Reply:
I always wanted to try uv actually. And we're really coming to a point now where that would make sense. We can let it set up a virtual encounter automatically with a python we want. As you said, rather in a next PR. I think we were always a bit reluctant to add more python specific infrastructure to the project, tools you need to know etc, since it's not _really_ used for prod code.

Reply:
Also we're soon having the packaging of the python bindings in here. Doesn't get more pythonically involved than that😄